projects/plugins/boost/app/lib/class-site-urls.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Lib;
+
+class Site_Urls {
+
+	public static function get( $limit = 1000 ) {
+		$instance = new static();
```

Comment:
I don't see any reason to instantiate the class, and then throw away the object it creates. If I'm reading this correctly, every member of this class could be static.

Reply:
Note: @pyronaur and I have a long-standing argument about whether purely-static classes have any value. I know that his position will be that this should be a collection of functions in a namespace, not a class.

My counter-argument is that using a class here allows you to have private members which are not exported into the namespace, keeping it cleaner.

~But~ I don't want Nauris and my eternal debates on clean code to slow development down. We fuss on details to the point that everyone else in the room falls asleep, so don't let our bickering over details bother you. Either an all-static class, or a bunch of standalone methods in an appropriately named namespace is fine, and will get this PR merged. ;)

I just don't think that we need to instantiate anything here, as no member variables are ever created or read.

Reply:
Thanks for the notes!

Initially, every method was static but also public. @pyronaur gave me some pointers and had a good argument about having only one entry point for the class.

> no member variables are ever created or read.

Since this was the reasoning behind my original approach, I updated all methods to static and kept them private (@pyronaur will be happy about the entry point 😅).

I'm not taking sides, just considering both approaches and my initial idea. After three iterations - I'm okay with how it turned out.